File: 496-next-greater-element-i/496-next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1, nums2):
        res = []
        for i in nums1:
            flag = 0
            _ = nums2.index(i)
            for j in nums2[_ + 1:]:
                if j > i:
                    res.append(j)
                    flag = 1
                    break

            if flag == 0:
                res.append(-1)
        return res

File: 496-next-greater-element-i/496-next-greater-element-i.go
func nextGreaterElement(nums1 []int, nums2 []int) []int {
    var res []int
    flag := 0
    for i:=0;i<len(nums1);i++{
        flag = 0
        for j:=0;j<len(nums2);j++{
            if nums1[i] == nums2[j]{
                for z:=j+1;z<len(nums2);z++{
                    if nums2[z] > nums1[i]{
                        flag = 1
                        res = append(res, nums2[z])
                        break
                    }
                }
                
            }
        }
        if flag == 0{
                res = append(res, -1)
            }
    }
    return res
}